doc/http-backoff.md

```diff
@@ -0,0 +1,10 @@
+# HTTP retry and backoff logic
+
+When Ignition is fetching a resource over http(s), if the resource is unavailable Ignition will continually retry to fetch the resource with an exponential backoff between requests. Once enough time has passed, Ignition will cancel the attempt and fail.
+
```

Comment:
If we're targeting having user docs in addition to the schema, I'd want an example of a valid `timeout` block here